Tests/Algorithm/Framework/Execution/StandardDeviationExecutionModelTests.cs

```diff
@@ -0,0 +1,151 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Moq;
+using NodaTime;
+using NUnit.Framework;
+using Python.Runtime;
+using QuantConnect.Algorithm.Framework;
+using QuantConnect.Algorithm.Framework.Execution;
+using QuantConnect.Algorithm.Framework.Portfolio;
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+using QuantConnect.Data.UniverseSelection;
+using QuantConnect.Interfaces;
+using QuantConnect.Orders;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Tests.Algorithm.Framework.Execution
+{
+    [TestFixture]
+    public class StandardDeviationExecutionModelTests
+    {
+        [TestFixtureSetUp]
+        public void SetUp()
+        {
+            PythonHelper.SetDefaultPythonPath();
+        }
+
+        [TestCase(Language.CSharp)]
+        [TestCase(Language.Python)]
+        public void OrdersAreNotSubmittedWhenNoTargetsToExecute(Language language)
+        {
+            var actualOrdersSubmitted = new List<SubmitOrderRequest>();
+
+            var orderProcessor = new Mock<IOrderProcessor>();
+            orderProcessor.Setup(m => m.Process(It.IsAny<SubmitOrderRequest>()))
+                .Returns((OrderTicket)null)
+                .Callback((SubmitOrderRequest request) => actualOrdersSubmitted.Add(request));
+
+            var algorithm = new QCAlgorithmFramework();
+            algorithm.SetPandasConverter();
+            algorithm.Transactions.SetOrderProcessor(orderProcessor.Object);
+
+            var model = GetExecutionModel(language);
+            algorithm.SetExecution(model);
+
+            var changes = new SecurityChanges(Enumerable.Empty<Security>(), Enumerable.Empty<Security>());
+            model.OnSecuritiesChanged(algorithm, changes);
+
+            model.Execute(algorithm, new IPortfolioTarget[0]);
+
+            Assert.AreEqual(0, actualOrdersSubmitted.Count);
+        }
+
+        [TestCase(Language.CSharp, new[] { 270d, 260d, 250d }, 1, 10)]
+        [TestCase(Language.CSharp, new[] { 250d, 250d, 250d }, 0, 0)]
+        [TestCase(Language.Python, new[] { 270d, 260d, 250d }, 1, 10)]
+        [TestCase(Language.Python, new[] { 250d, 250d, 250d }, 0, 0)]
+        public void OrdersAreSubmittedWhenRequiredForTargetsToExecute(
+            Language language,
+            double[] historicalPrices,
+            int expectedOrdersSubmitted,
+            decimal expectedTotalQuantity)
+        {
+            var actualOrdersSubmitted = new List<SubmitOrderRequest>();
+
+            var i = 1;
+            var time = new DateTime(2018, 8, 2, 16, 0, 0);
+            var historyProvider = new Mock<IHistoryProvider>();
+            historyProvider.Setup(m => m.GetHistory(It.IsAny<IEnumerable<HistoryRequest>>(), It.IsAny<DateTimeZone>()))
+                .Returns(historicalPrices.Select(x =>
+                    new Slice(time.AddMinutes(i++),
```

Comment:
no change required, but fyi, `Select` has an overload that accepts a `Func<T, int, TResult>` and can be used like:
``` csharp
var slices = historicalPrices.Select( (x,i) => new Slice(time.addMinutes(i));
```

This ensures that our LINQ expression doesn't depend on side effects (LINQ is a functional library and side effects are the antithesis of the functional mindset :) )